pkg/server/watch.go

```diff
@@ -39,7 +38,19 @@ func (s *KVServerBridge) Watch(ws etcdserverpb.Watch_WatchServer) error {
 
 	logrus.Tracef("WATCH SERVER CREATE")
 
-	go wait.PollInfiniteWithContext(ws.Context(), s.getProgressReportInterval(), w.ProgressIfSynced)
+	go func(ctx context.Context) {
+		ticker := time.NewTicker(s.getProgressReportInterval())
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ws.Context().Done():
+				return
+			case <-ticker.C:
+				w.ProgressIfSynced(ctx)
+			}
+		}
+	}(ws.Context())
```

Comment:
ugh I hate how ugly this inline func is compared to the wait helpers. I think we used to have something similar to this and I replaced it with the wait.

Reply:
Well the wait utility function itself is deprecated. Would you rather keep the K8s dependency, but move to the newer wait function, the `PollUntilContextCancel`?

Reply:
Maybe a utility function in https://github.com/k3s-io/kine/tree/master/pkg/util? idk I just hate sprinkling goroutine ticker loop functions around otherwise tidy functions.